app/components/storage/storage.service.js

```diff
@@ -22,24 +24,22 @@
       return (video) ? video.dateAdd : false;
     }
 
-    function addVideo(video) {
+    function addItem (video) {
+      let defer = $q.defer();
       let videoList = getVideoList();
-      let newObj = {
-        id: video.id.videoId,
-        type: video.id.kind,
-        name: video.snippet.title,
-        thumbnails: video.snippet.thumbnails,
-        description: video.snippet.description,
+      let baseFields = {
         dateAdd: moment().toDate().getTime(),
         fav: false
       };
 
       // @todo pass thru checkfordupes func
-      videoList.push(newObj);
-      updateVideoList(videoList);
+      videoList.push(Object.assign(video, baseFields));
+      defer.resolve(updateVideoList(videoList));
```

Comment:
uzywasz promise'ow zle.
https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern

dodatkowo przeczytaj
https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html